R/fillsink.R

```diff
@@ -0,0 +1,36 @@
+#' wrap_fillsink
+#'
+#' This will make the fillsinks funktion available to R from the libtotopotoolbox subdirectory
+#' 
+#' @param dem Takes a Spatrast as input 
+#' @param bc Array of the same size as the input matrix
+#'
+#' @return A Spatrast
+#' @export
+
+fillsink <- function(dem,bc=NULL) {
+    
+    # if user didn't set bc, then set it manual
+    if (is.null(bc))
+    {
+        print("ich bin da")
+        bc <- matrix(0,dim(dem)[1],dim(dem)[2])
```

Comment:
This needs to have the dimensions flipped: `bc <- matrix(0,dim(dem)[2],dim(dem)[1])` because R stores its arrays in row-major format, but libtopotoolbox expects the arrays in row-major format.

Alternatively, you could move the `d <- get_grid_data(dem)` line above where you create `bc`, and use `matrix(0,d$dims[1],d$dims[2])` because we already flip the dimensions in `get_grid_data`.